### PR TITLE
Update to GTK3 build with clutter and Gnome 42 [Testing]

### LINKED ISSUE #16: GTK3 Disabled?
I notice in b7ce1ded11c0aa35dc19dc63d0f34f74066b5fdd the GTK version was downgraded from 3 to 2. GTK3 would provide native Wayland support. Is there a particular reason it was removed?

### PR DESCRIPTION
**Pull request by yourealwaysbe**
Here is my first effort of a build for GTK3. I have tested only briefly and it looks ok, but it would be worth checking the errors that led to GTK3 being removed are gone.

I updated to Gnome 42, perhaps that was premature.

I had to update the shared-modules to get access to the clutter library needed by libchamplain. Added the clutter library to the build.

Fixes #16

### CODE REVIEW

File: org.geeqie.Geeqie.json [hunk 1]
```diff
@@ -94,21 +95,21 @@
...
-                "--disable-gtk3"
+                "--enable-gtk3"
             ],
             "cleanup": [
                 "/share/man"
             ],
             "post-install": [
-                "install -Dm644 src/icons/svg/gqview_icon.svg /app/share/icons/hicolor/scalable/apps/geeqie.svg",
+                "install -Dm644 org.geeqie.Geeqie.appdata.xml /app/share/metainfo/org.geeqie.Geeqie.appdata.xml",
+                "install -Dm644 /app/share/pixmaps/geeqie.png /app/share/icons/hicolor/48x48/apps/geeqie.png",
```
Comment (Hofer-Julian) [inline org.geeqie.Geeqie.json:112]: Why did you change the icon?

Comment (yourealwaysbe) [reply org.geeqie.Geeqie.json:112]: I was naively undoing the commit from GTK3 -> GTK2. Will change that. Thanks.

**Comment by yourealwaysbe** [comment]
I have tested the sidebar bug mention in #4 

If you change the size of the sidebar and restart, position is maintained, but only within bounds. I think there's a min width (to stop elements not displaying fully) and a max width (about 1/3 of the screen width).

**Comment by Hofer-Julian** [comment]
> I have tested the sidebar bug mention in #4
> 
> If you change the size of the sidebar and restart, position is maintained, but only within bounds. I think there's a min width (to stop elements not displaying fully) and a max width (about 1/3 of the screen width).

I think you are right with your assumption that Geeqie enforces a minimum width for the sidebar when compiled with GTK3.
Not going to block this PR because of that, but would you expect the behaviour shown in the video @caclark?


[Kooha-07-08-2022-14-42-08.webm](https://user-images.githubusercontent.com/30049909/177994265-90bbeee4-78d3-42e7-b23a-dfb4ce86261f.webm)

**Review by Hofer-Julian — APPROVED** [review]

**Comment by caclark** [comment]
> but would you expect the behaviour shown in the video @caclark?

When Geeqie shuts down, the position of the windows is stored. In this case the function gtk_widget_get_allocation() is used to get the current layout.
This function behaves in a different way in GTK2 and in GTK3.
I have tried other functions, intercepting the "size-allocate" signal, all to no avail.
If anyone knows of a solution, that would be good to know about.

[I see that GTK4 has a new function - gtk_widget_get_size(). Maybe that will solve the problem]

**Comment by Hofer-Julian** [comment]
Thanks for the context!
